lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/MavenLemminxExtension.java

```diff
@@ -467,4 +469,8 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
 
 		return projectsToAdd;
 	}
+	
+	private void registerCodeActionParticipants(XMLExtensionsRegistry registry) {
+		registry.registerCodeActionParticipant(new MavenNoGrammarConstraintsCodeAction());

```

Comment:
Those need to be unregistered on stop. I suggest you keep the list of registered codeActionParticipants as a field so you can then call `codeActionParticipants.forEach(registry::unregisterCodeActionParticipant)` in `stop`

Reply:
We have the list of all registered code action participants in `registry.getCodeActionsParticipants()` so, no need to keep yet another list during the working session, no?

Reply:
Used a local list of participants in order to not un-register the participants from the other extensions